### PR TITLE
Fix the service_controller test cases and some syntax errors

### PR DESCRIPTION
**Pull request by ping035627**
**What this PR does:**
/kind cleanup
Fix the service_controller test cases and some syntax errors

 **Why we need it:**
Perfect the test cases and correct syntax errors

**Which issue(s) this PR fixes** (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):
Fixes #

**Special notes for your reviewer:**

**Release note:**

NONE

### CODE REVIEW

**Comment by ping035627** [comment]
I signed it

**Comment by ping035627** [comment]
I signed it

**Comment by ping035627** [comment]
I signed it

File: pkg/controller/service/service_controller_test.go [hunk 1]
```diff
@@ -325,8 +325,6 @@ func TestGetNodeConditionPredicate(t *testing.T) {
 	}
 }
 
-// TODO(a-robinson): Add tests for update/sync/delete.
```
[the remark this answers is not in the document]
Comment (ping035627) [reply pkg/controller/service/service_controller_test.go]: I think it refers to TestProcessServiceUpdate/TestConflictWhenProcessServiceUpdate/TestSyncService/TestProcessServiceDeletion, these have been done, so I delete the TODO.

File: pkg/controller/service/service_controller_test.go [hunk 2]
```diff
@@ -462,11 +460,9 @@ func TestSyncService(t *testing.T) {
 			key:      "invalid/key/string",
 			updateFn: func() {
 				controller, _, _ = newController()
-
 			},
 			expectedFn: func(e error) error {
-				//TODO: Expected error is of the format fmt.Errorf("unexpected key format: %q", "invalid/key/string"),
```
[the remark this answers is not in the document]
Comment (ping035627) [reply pkg/controller/service/service_controller_test.go]: I have added this judgement condition, please review it again, thanks.

**Comment by ping035627** [comment]
@andrewsykim hi, can you help to review this pr? thanks.

File: pkg/controller/service/service_controller_test.go [hunk 3]
```diff
@@ -762,16 +760,16 @@ func TestServiceCache(t *testing.T) {
 		{
 			testName: "Set and Get",
 			setCacheFn: func() {
-				sc.set("addTest", &cachedService{state: defaultExternalService()})
```
Comment (andrewsykim) [inline pkg/controller/service/service_controller_test.go]: I'm wondering if it make sense for this to be shared across all the test tables? It seems like we're leaking state across test runs.

Comment (ping035627) [reply pkg/controller/service/service_controller_test.go]: Oh, I see, I have returned it. I just think it is not rigorous to judge empty only.

**Comment by ping035627** [comment]
@andrewsykim excuse me, can you review this pr again? thanks.

**Comment by andrewsykim** [comment]
/approve
/lgtm

Thanks @ping035627

**Comment by ping035627** [comment]
@andrewsykim Does "do-not-merge/release-note-label-needed" label should be removed?

**Pull request merged**